File: 18. Backend Scaling and Performance Engineering-part-1/code/typescript/example_7.ts
import express from 'express';
import { Pool } from 'pg';

const app = express();
const db = new Pool();

app.get('/health', async (req, res) => {
    try {
        await db.query('SELECT 1');
        res.status(200).json({ status: 'healthy' });
    } catch (err) {
        res.status(503).json({ status: 'unhealthy', reason: 'database unreachable' });
    }
});

File: 18. Backend Scaling and Performance Engineering-part-1/code/rust/example_7.rs
use actix_web::{get, web, HttpResponse, Responder};
use sqlx::PgPool;

#[get("/health")]
async fn health(db: web::Data<PgPool>) -> impl Responder {
    match sqlx::query("SELECT 1").execute(db.get_ref()).await {
        Ok(_) => HttpResponse::Ok().json(serde_json::json!({ "status": "healthy" })),
        Err(_) => HttpResponse::ServiceUnavailable().json(serde_json::json!({
            "status": "unhealthy",
            "reason": "database unreachable"
        }))
    }
}
